docs/transfer.md

```diff
@@ -0,0 +1,32 @@
+# Transfer Service
+
+The transfer service is a simple flexible service which can be used to transfer artifact objects between a source and destination. The service determines whether the transfer between the source and destination is possible rather than the API.
+
+The transfer service if built upon the core ideas put forth by the libchan project, that an API with binary streams and data channels as first class objects is more flexible and opens a wider variety of use cases without requiring constant protocol and API updates. To accomplish this, the transfer service makes use of the streaming service to allow binary and object streams to be accessible by transfer objects even when using grpc and ttrpc.
+
+
+## Transfer Objects (Sources and Destinations)
+
+## Transfer Operations
+
+|   Source    | Destination | Description |
+|-------------|-------------|-------------|
+| Registry    | Image Store | "pull"      |
+| Image Store | Registry    | "push"      |
+| Object stream (Archive) | Image Store | "import" |
+| Image Store | Object stream (Archive) | "export" |
+| Object stream (Layer) | Mount/Snapshot | "unpack" |
+| Mount/Snapshot | Object stream (Layer) | "diff" |
+| Image Store | Image Store | "tag" |
```

Comment:
Does retagging an image involve using the transfer service?

Reply:
You could, today tagging is two API calls, Get and Create. The point is highlighting the flexibility. We will never have a `Tag` API call, but you could do the equivalent using the image API or also the transfer API. The transfer API could be more useful if the image stores have separate content backends or in the case where one of the image stores was actually inside a shim or some other sandboxed environment.